contrib/registry/consul/registry.go

```diff
@@ -135,7 +135,7 @@ func (r *Registry) Watch(ctx context.Context, name string) (registry.Watcher, er
 	}
 
 	if !ok {
-		go r.resolve(set)
+		r.resolve(set)
```

Comment:
If resolve fails the first time, an error should be returned.

Reply:
Yes, you are right.